src/workerd/api/node/zlib-util.h

```diff
@@ -291,17 +291,23 @@ class ZlibUtil final: public jsg::Object {
   ZlibUtil() = default;
   ZlibUtil(jsg::Lock&, const jsg::Url&) {}
 
-  // A custom allocator to be used by the zlib and brotli libraries
-  // The current implementation stores allocations in a hash map.
-  // TODO: Use an arena allocator implementation instead of hashing pointers in order to improve performance
+  // A custom allocator to be used by the zlib and brotli libraries.
+  // The allocator should not and can not safely hold a reference to the jsg::Lock
+  // instance. Therefore, we lookup the current jsg::Lock instance from the
+  // isolate pointer and use that to get the external memory adjustment.
   class Allocator final {
   public:
     static void* AllocForZlib(void* data, uInt items, uInt size);
     static void* AllocForBrotli(void* data, size_t size);
     static void FreeForZlib(void* data, void* pointer);
 
+    struct Allocation {
+      kj::Array<kj::byte> data;
+      kj::Maybe<jsg::ExternalMemoryAdjustment> memory;
```

Comment:
```suggestion
      kj::Maybe<jsg::ExternalMemoryAdjustment> memoryAdjustment;
```

nit: Makes it clearer which field is the allocation and what is the metadata.